### DISCUSSION TITLE
Smart Watering Moisture Level Tracking

### DISCUSSION THREAD

**Discussion by ChipSkylark37** [sebr/bhyve-home-assistant#206] [Q&A]
Hi all,

I was reading through a few issues, and see that there may be a way to set the state / value of the moisture levels of zones. 

Is there a way to read the current moisture level of zones when using smart watering? (Or have this in the history / logbook) Specifically, b-hyve is not always setting the moisture levels to 100% after watering causing the system to water repeatedly and overwatering, so I'd like to be able to track the moisture level before and after smart watering. If the new value = the old value, that would indicate a problem. 

I've checked though some of the attributes for zones, and don't see that it's exposed. 

This is the specific model I have but assume it would also be for any one that stores the data in the orbit cloud account: Orbit 57946 B-hyve

**Comment by sebr** [comment]
> Is there a way to read the current moisture level of zones when using smart watering?

Yes, see https://github.com/sebr/bhyve-home-assistant/blob/main/custom_components/bhyve/switch.py#L637

Soil moisture levels are not sent on any of the events as far as I know and it would be necessary to create a new Sensor which periodically queries the `get_landscape` API call.

You can follow [`BHyveZoneHistorySensor`](https://github.com/sebr/bhyve-home-assistant/blob/main/custom_components/bhyve/sensor.py#L187) as a pattern which polls the API.
